specification/hybridkubernetes/resource-manager/Microsoft.Kubernetes/preview/2022-10-01-preview/connectedClusters.json

```diff
@@ -639,6 +656,28 @@
         "privateLinkScopeResourceId": {
           "type": "string",
           "description": "The resource id of the private link scope this connected cluster is assigned to, if any."
+        },
+        "azureHybridBenefit": { 
+          "type": "string", 
+          "description": "Indicates whether Azure Hybrid Benefit is opted in",
+          "enum": [
+            "True",
+            "False",
+            "NotApplicable"
+          ],
+          "default": "NotApplicable",
+          "x-ms-enum": {
+            "name": "AzureHybridBenefit",
+            "modelAsString": true
+          }
+        },
+        "additionalProperties": {
```

Comment:
This looks a little confusing. Normally additionalProperties (reserved property) would be of type object/boolean. If your intention is to provide flexibility of allowing a dictionary of properties for posterity, then you can just maintain string->string mapping.. Or to be even more extensible. string->object(string>string) mappings.. That should do.. 

Something like this - 
################################
          "type": "object",
          "additionalProperties": {
            "type": "string"
          },
################################
Reference - https://swagger.io/specification/#:~:text=For%20a%20simple%20string%20to%20string%20mapping%3A

Reply:
Modified as per the review comment.

Reply:
ARM recommends against the use of the additionalProperties/empty/unschematized object pattern. ARM favors stronger typing wherever possible because it allows all consumers of your API to reason about it. This has benefits for everyone that interacts with your API including basic infrastructure, other Azure features, SDK owners, and customers. Unschematized objects are essentially API contracts that bypass Azure versioning infrastructure.

For example, customers will not be able to manage these objects using Azure Policy, or query them using ARG, both of which are standard Azure features. Customers expect to be able to use all common Azure features against all of their resources. 

Generally, this pattern should only be used when needed and the object properties are dynamic, unknown, or user defined. Ref: [https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined](https://armwiki.azurewebsites.net/api_contracts/guidelines/openapi.html#oapi032-only-use-additionalproperties-when-the-object-properties-are-dynamic-unknown-or-user-defined).

Reply:
We want to expose a property bag/dictionary for properties that some of our partners can use. 
The values that our partners will store in this property bag/dictionary are dynamic and their keys are not known, hence the use of additionalProperties.